docs/Testing.md

```diff
@@ -1,13 +1,157 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap pino-pretty --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+app.js
+
+```js 
+const Fastify = require('fastify')
+
+function build(opts) {
+  const fastify = Fastify(opts)
+  fastify.get('/', async (request, reply) => {
+    return { hello: 'world' }
+  })
+
+  return fastify
+}
+
+module.exports = build
+```
+
+server.js
+
+```js
+const server = require('./app')({
+    logger: {
+      level: 'info',
+      prettyPrint: true
+    }
+  })
+
+server.listen(3000, (err, address) => {
+  if (err) {
+    console.log(err)
+    process.exit(1)
+  }
+  console.log(`server listening on ${address}`)
+})
+```
+
+### Benefits of using fastify.inject()
+
 Fastify comes with built-in support for fake http injection thanks to [`light-my-request`](https://github.com/fastify/light-my-request).
 
-To inject a fake http request, use the `inject` method:
+Before introducing any tests, we'll use the `.inject` method to make a fake request to our route:
+
+app.test.js
+
+```js
+const build = require('./app')
+
+const runTests = async () => {
+  const fastify = build({
+    logger: {
+      level: 'info',
+      prettyPrint: true
+    }
+  })
+  await fastify.ready()
+  
+  try {
+  const response = await fastify.inject(
+    {
+      method: 'GET',
+      url: '/'
+    }
+  )
+  console.log('status code: ', response.statusCode)
+  console.log('body: ', response.body)
+} catch (err) {
+  // handle error
+}
```

Comment:
Please remove the try-catch and wrap `runTests` in the `test` function from tap.